x/vote/types/types.go

```diff
@@ -192,21 +192,23 @@ func (p *Poll) Vote(voter sdk.ValAddress, blockHeight int64, data codec.ProtoMar
 		return nil, false, nil
 	case p.Is(exported.Expired):
 		return nil, false, fmt.Errorf("poll %s has expired already", p.GetKey().String())
-	case p.Is(exported.Completed) && p.isWithinGracePeriod(blockHeight):
+	case p.Is(exported.Completed):
 		votingPower := p.getVotingPower(voter)
 		if votingPower == 0 {
```

Comment:
Check this in both cases to not allow non-participants to submit late votes. This occurs for validators who were auto deregistered (and thus still have the vald bridge enabled) and end up submitting no-op [late votes](https://axelarscan.io/evm-votes?pollId=%220xcfac1b643967d988914a88dfd67feabb37093552ed4079f62cc57c54775f4eec_0xFE6B3cFc6474d1BeC22800317796a16103722300%22).

Reply:
Why not move the `votingPower` check then out to always be checked before anything?